opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -402,8 +412,13 @@ impl PeriodicReaderInner {
     }
 
     fn shutdown(&self) -> MetricResult<()> {
-        if self.is_shutdown.load(std::sync::atomic::Ordering::Relaxed) {
-            return Err(MetricError::Other("Reader is already shut down".into()));
+        if self
```

Comment:
Could we just use a `Mutex` object to synchronize `shutdown` and `force_flush` calls? That would greatly simplify things by eliminating both the race conditions.